### ISSUE TITLE
Explicit `serializer_lookup_enabled` in controller doesn't retain Adapter for entire render 

### ISSUE THREAD

**Issue by trek** [rails-api/active_model_serializers#1500]
There's some wonky interaction in #1353 that was not present in #1295:

I've isolated this to a integration example: https://github.com/trek/ams-nesting-test

To run the test:

```
git clone https://github.com/trek/ams-nesting-test.git
cd ams-nesting-testing
bundle
bundle exec rake db:migrate
bundle exec rspec spec/controllers
```

The serializer is configured to use JSON API and require an explicit `serializer` option in renders.

https://github.com/trek/ams-nesting-test/blob/e2d590f79a68b76299003e4021499c34544040cc/config/initializers/active_model_serializers.rb

That feature is used here:

https://github.com/trek/ams-nesting-test/blob/e2d590f79a68b76299003e4021499c34544040cc/app/controllers/tweets_controller.rb#L4

You would expect, in [this test case](https://github.com/trek/ams-nesting-test/blob/e2d590f79a68b76299003e4021499c34544040cc/spec/controllers/tweets_controller_spec.rb) to see a JSON response of

```
{
  "data": [
    {
      "id": "1",
      "type": "tweets",
      "attributes": {
        "body": "lolz"
      },
      "relationships": {
        "favorites": {
          "data": [
            {
              "id": "1",
              "type": "favorites"
            },
            {
              "id": "2",
              "type": "favorites"
            }
          ]
        }
      }
    }
  ],
  "included": [
    {
      "id": "1",
      "type": "favorites",
      "attributes": {
        "reason": "one reason"
      }
    },
    {
      "id": "2",
      "type": "favorites",
      "attributes": {
        "reason": "another reason"
      }
    }
  ]
}
```

but instead see

```
{
  "data": [
    {
      "id": "1",
      "type": "tweets",
      "attributes": {
        "body": "lolz"
      },
      "relationships": {
        "favorites": {
          "data": [
            {
              "id": 1,
              "tweet_id": 1,
              "reason": "one reason",
              "created_at": "2016-02-06T17:34:57.611Z",
              "updated_at": "2016-02-06T17:34:57.611Z"
            },
            {
              "id": 2,
              "tweet_id": 1,
              "reason": "another reason",
              "created_at": "2016-02-06T17:34:57.615Z",
              "updated_at": "2016-02-06T17:34:57.615Z"
            }
          ]
        }
      }
    }
  ]
}
```

where `relationship` values appear to fall back to the `attributes` serializer.

If you _also_ explicitly state which serializer to use for the include association, you do see the correct output:

```
class TweetSerializer < ActiveModel::Serializer
  attribute :body

  # does not work
  # has_many :favorites
  
  # works
  has_many :favorites, serializer: FavoriteSerializer
end
```

**Comment by bf4** [comment]
@trek thanks for this.  So, I see your original case was that you essentially wanted to disable AMS  in the controller unless it was specifically asked for by passing an option, but I got focused on the lookup idea and missed the scope or your use-case

>   # does not work
>   # has_many :favorites
> 
>   # works
>   has_many :favorites, serializer: FavoriteSerializer

This was intended behavior on my part, but not really useful for you, since then the object with an unspecified serializer uses its own `to_json` method that the JSON gem and ActiveSupport give it...

So, the real fix for your case would be `config.controller_requires_serializer_option = true` or something like that which defaults to `false`. Does that sound about right?

**Comment by trek** [comment]
@bf4 yep, yep. The original use case was opting into AMS per `render` call because we have lots of old controllers whose original behavior we want to maintain.

Once a `render` call says to use AMS, we want it to be AMS all the way down.

`config.controller_requires_serializer_option` seems good.

**Comment by bf4** [comment]
@trek wanna do the honors?

**Comment by bf4** [comment]
ref: https://github.com/rails-api/active_model_serializers/issues/1442

**Comment by bf4** [comment]
In 0.9

- https://github.com/rails-api/active_model_serializers/pull/592
  - https://github.com/rails-api/active_model_serializers/pull/593
  - https://github.com/rails-api/active_model_serializers/pull/611

```ruby
# ActionController:Serialization
    class << self
       attr_accessor :enabled
     end
     self.enabled = true
end
# Railtie
  ActiveSupport.on_load(:action_controller) do
      if ::ActionController::Serialization.enabled
        ActionController::Base.send(:include, ::ActionController::Serialization)
```

which I think I'll handle via

`Rails.configuration.action_controller.perform_active_model_serialization` so our Railtie can ensure it's taken care of
